pkg/controller/deployment/deployment_controller.go

```diff
@@ -410,6 +411,16 @@ func (dc *DeploymentController) syncDeployment(key string) error {
 		return nil
 	}
 
+	if d.Spec.Rollback != nil {
+		revision := d.Spec.Rollback.Revision
+		if err := dc.rollback(d, revision); err != nil {
+			glog.Infof("Unable to rollback deployment %v to revision %d: %v", key, revision, err)
+		} else {
+			// clear rollback when we're done
+			dc.clearRollback(&d)
```

Comment:
We should be able to revert the pod template, update the revision annotation, and clear the rollback config in a single update to the Deployment.